core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -361,10 +361,12 @@ private CloseableIterable<ManifestEntry<DataFile>> addedDataFiles(TableMetadata
    * @param base table metadata to validate
    * @param startingSnapshotId id of the snapshot current at the start of the operation
    * @param dataFiles data files to validate have no new row deletes
+   * @param ignorePosDeletes whether position deletes should be ignored in validation
    */
   protected void validateNoNewDeletesForDataFiles(TableMetadata base, Long startingSnapshotId,
-                                                  Iterable<DataFile> dataFiles) {
-    validateNoNewDeletesForDataFiles(base, startingSnapshotId, null, dataFiles, newFilesSequenceNumber != null);
+                                                  Iterable<DataFile> dataFiles, boolean ignorePosDeletes) {
+    validateNoNewDeletesForDataFiles(base, startingSnapshotId, null, dataFiles, newFilesSequenceNumber != null,
+        ignorePosDeletes);
```

Comment:
It doesn't seem to make sense to make one flag explicit and the other one implied by the sequence number.

Reply:
It does look weird, let me refactor this.